src/core/js/models/adaptModel.js

```diff
@@ -287,24 +289,42 @@ define([
                 }
             }
 
-            searchChildren(children);
+            if (this._children === descendants) {
+                returnedDescendants = this.getChildren().models;
+            } else {
+                searchChildren(this.getChildren().models);
+            }
 
-            return returnedDescendants;
+            if (!options) {
+                return returnedDescendants;
+            }
+
+            if (options.where) {
```

Comment:
Please can we return early:
```js
var properties = options.where;

if (!properties) return;
```

Reply:
@tomgreenfield the reason I didn't return early here is that there's an expectation that further `options` will be added in future - see [this suggestion](https://github.com/adaptlearning/adapt_framework/issues/2058#issuecomment-385449881) from @oliverfoster